modules/gdscript/doc_classes/@GDScript.xml

```diff
@@ -77,8 +77,12 @@
 		<method name="dict_to_inst">
 			<return type="Object" />
 			<param index="0" name="dictionary" type="Dictionary" />
+			<param index="1" name="depth" type="int" default="0" />
 			<description>
 				Converts a [param dictionary] (created with [method inst_to_dict]) back to an Object instance. Can be useful for deserializing.
+				When [param depth] is [code]0[/code] only one level of values of [param dictionary] will be converted to an Object instance
+				When [param depth] is [code]x > 0[/code] only [code]x[/code] levels of values of [param dictionary] will be converted to nested Object instances
+				When [param depth] is [code]-1[/code] all nested values of [param dictionary] will be converted to nested Object instances
```

Comment:
I wouldn't recommend allowing infinite recursion, otherwise we might end up with something like #55256. We probably need to replace `depth: int` with `deep: bool` and use a hardcoded ([`MAX_RECURSION`](https://github.com/search?q=repo%3Agodotengine%2Fgodot%20max_recursion&type=code)) value.